airflow/models/dag.py

```diff
@@ -905,22 +905,24 @@ def get_num_active_runs(self, external_trigger=None, session=None):
         return query.scalar()
 
     @provide_session
-    def get_dagrun(self, execution_date, session=None):
+    def get_dagrun(self, execution_date: str = None, run_id: str = None, session=None):
```

Comment:
If the argument could be None, the type should be `Optional[str]`. (In this particular case, we can actually set the default to `""` since both the execution date and run_id cannot be empty, but that might not be as intuitive to read.)